azurerm/resource_arm_netapp_pool.go

```diff
@@ -0,0 +1,220 @@
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, accountName, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for present of existing NetApp Pool %q (Resource Group %q): %+v", name, resourceGroup, err)
+			}
+		}
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_netapp_pool", *existing.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	serviceLevel := d.Get("service_level").(string)
+	size := int64(d.Get("size").(int))
+
+	capacityPoolParameters := netapp.CapacityPool{
+		Location: utils.String(location),
+		PoolProperties: &netapp.PoolProperties{
+			ServiceLevel: netapp.ServiceLevel(serviceLevel),
+			Size:         utils.Int64(size),
+		},
+	}
+
+	future, err := client.CreateOrUpdate(ctx, capacityPoolParameters, resourceGroup, accountName, name)
+	if err != nil {
+		return fmt.Errorf("Error creating NetApp Pool %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for creation of NetApp Pool %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, accountName, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving NetApp Pool %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read NetApp Pool %q (Resource Group %q) ID", name, resourceGroup)
+	}
+	d.SetId(*resp.ID)
+
+	return resourceArmNetAppPoolRead(d, meta)
+}
+
+func resourceArmNetAppPoolRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Netapp.PoolClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	accountName := id.Path["netAppAccounts"]
+	name := id.Path["capacityPools"]
+
+	resp, err := client.Get(ctx, resourceGroup, accountName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] NetApp Pools %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading NetApp Pools %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("account_name", accountName)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	if poolProperties := resp.PoolProperties; poolProperties != nil {
+		if err := d.Set("service_level", poolProperties.ServiceLevel); err != nil {
+			return fmt.Errorf("Error setting `service_level`: %+v", err)
+		}
+
+		if err := d.Set("size", poolProperties.Size); err != nil {
+			return fmt.Errorf("Error setting `size`: %+v", err)
+		}
```

Comment:
same for datasource, why the err checks?

Reply:
Fixed